### ISSUE TITLE
Collapsible elements need aria-controls and aria-expanded [work on PR #14880/Windows]

### ISSUE THREAD

**Issue by VGRAJED** [SAP/spartacus#11637]
Product tabs in Product Details Page and Product facets in Product Listing Page need aria attributes such as aria-controls and aria-expanded.

Probably a service and directive will be helpful to track state of aria-expanded

**Comment by developpeurweb** [comment]
Adding only `aria-controls` and `aria-expanded` will improve the current solution, but will not suffice for full Screen Reader support. It should be complemented with `role`, `aria-label` and `aria-selected`.

My recommendation is to change the Accessible Pattern of both components to fully align with ARIA best practices regarding Keyboard Support and correct Screen Reader narration.

**Roles definition and meaningful labelling are crucial in both cases.**

 ## PDP Tab group

**ARIA Tab group**
https://www.w3.org/TR/wai-aria-practices/examples/tabs/tabs-2/tabs.html

**Notes:**
- The current arrangement switches from Tabs to Accordions in mobile.
- Future designs of this area contemplates an Accordion arrangement.
- Considering the previous statements, this makes it hard to change the accessible pattern. The ideal solution would be if we could use all 4 arrow keys and **make it work for both patterns**.


## PLP Facets

Combine ARIA design patterns for both, **Accordion** and **Checkbox**

**ARIA Accordion:**
https://www.w3.org/TR/wai-aria-practices-1.1/examples/accordion/accordion.html

**ARIA Checkbox:**
https://www.w3.org/TR/wai-aria-practices-1.1/examples/checkbox/checkbox-1/checkbox-1.html


**Notes:**
- Current checkboxes are in fact links, they need a `role` and `aria-checked`.
- Currently, the focus lock works by hitting RETURN on the selected group once and then ESC 2x to unlock. This is not a known pattern and falls into the "Custom Widget" category, therefore descriptions and instructions for Screen Readers users should be added otherwise by not communicating the 2x ESC trick the user will be trapped, or change the Accessible Pattern to a known pattern.
- Related ticket: GH-7180

**Comment by Xymmer** [comment]
best effort for 4.0, or put in 4.x; or split into 2 tickets

**Comment by Xymmer** [comment]
should we aim for 5.0 now for this one?

**Comment by developpeurweb** [comment]
While the agnostic a11y-tabs component continues to get developed, the Accordions/Tab issue still persists for both **Facet Search** and **PDP**.

 So, for now, in order to make both Facet Search and PDP work with SR and pass the audit, let's use accordions for everything. 

That is, **Facet Search** needs an accordion with an a11y pattern (same as you have on your POC: a keyboard pattern that vocalizes expanded/collapsed state) and the same for the **PDP** as per Manu's mockup below.

![Screen Shot 2022-01-24 at 11 30 07 AM](https://user-images.githubusercontent.com/15113219/150990774-a3e647a8-209e-4604-b13f-4bc69deb652d.png)

**Comment by FollowTheFlo** [comment]
From today investigation:
- The issue is only reproducible with NVDA on Chrome (OK on Firefox) 
 -> steps: on PDP, tab to an accordion item header, press space, nothing happens. It should expand or retract the item content.
- The issue is due to +(plus) char added via css (_tab-paragraph-container.css) with property "float: right;"
- 
![image](https://user-images.githubusercontent.com/30024415/155446557-4398093d-d846-4516-be6d-0a5ff1eb418c.png)

-> By removing "float:right", it fixes the issue but buttons appears before the title and not aligned
<img width="1147" alt="image" src="https://user-images.githubusercontent.com/30024415/155446861-bccb7232-8289-43e0-8e2a-e1bc38a41541.png">

Solution:
For minimum change -> a css alternative of 'float:right' to put +/- buttons on right side of accordion item header. I'll investigate further.
